autosubmit/notifications/mail_notifier.py

```diff
@@ -53,26 +65,34 @@ def notify_status_change(self, exp_id, job_name, prev_status, status, mail_to):
                 Log.printlog('Trace:{0}\nAn error has occurred while sending a mail for the job {0}'.format(e,job_name), 6011)
 
     def _send_mail(self, mail_from, mail_to, message):
-        server = smtplib.SMTP(self.config.SMTP_SERVER,timeout=60)
+        server = smtplib.SMTP(self.config.SMTP_SERVER, timeout=60)
         server.sendmail(mail_from, mail_to, message.as_string())
         server.quit()
 
+    def _attach_files(self, message, files):
+        for f in files or []:
+            with open(f, "rb") as file:
+                part = MIMEApplication(file.read(), Name = Path(f).name)
+                part['Content-Disposition'] = 'attachment; filename="%s"' % Path(f).name
+                message.attach(part)
+
+
     @staticmethod
     def _generate_message_text(exp_id, job_name, prev_status, status):
-        return f'Autosubmit notification\n' \
-               f'-------------------------\n\n' \
-                f'Experiment id:  {str(exp_id)} \n\n' \
-                + f'Job name: {str(job_name)} \n\n' \
-                f'The job status has changed from: {str(prev_status)} to {str(status)} \n\n\n\n\n' \
-                f'INFO: This message was auto generated by Autosubmit, '\
-                f'remember that you can disable these messages on Autosubmit config file. \n'
+        return f'''Autosubmit notification\n
+               -------------------------\n\n
+               Experiment id:  {str(exp_id)} \n\n
+               Job name: {str(job_name)} \n\n
+               The job status has changed from: {str(prev_status)} to {str(status)} \n\n\n\n\n
+               INFO: This message was auto generated by Autosubmit,
+               remember that you can disable these messages on Autosubmit config file. \n'''
 
     @staticmethod
     def _generate_message_experiment_status(exp_id, platform=""):
```

Comment:
Maybe `platform=""` needs to change? Can you change that to the correct default value if any (e.g. `None`?), and add type hints for mypy/ide/etc, please? We are adding these to new code as we add/modify. For example, something like,

```diff
diff --git a/autosubmit/notifications/mail_notifier.py b/autosubmit/notifications/mail_notifier.py
index bfcea3c6..4bdaff49 100644
--- a/autosubmit/notifications/mail_notifier.py
+++ b/autosubmit/notifications/mail_notifier.py
@@ -25,6 +25,11 @@ from email.mime.application import MIMEApplication
 from log.log import Log
 from pathlib import Path
 from autosubmitconfigparser.config.basicconfig import BasicConfig
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from autosubmit.platforms.platform import Platform
 
 class MailNotifier:
     def __init__(self, basic_config):
@@ -88,11 +93,11 @@ class MailNotifier:
                remember that you can disable these messages on Autosubmit config file. \n'''
 
     @staticmethod
-    def _generate_message_experiment_status(exp_id, platform=""):
+    def _generate_message_experiment_status(exp_id: str, platform: Platform = None) -> str:
         return f'''Autosubmit notification: Remote Platform issues\n-------------------------\n
-               Experiment id: {str(exp_id)}
+               Experiment id: {exp_id}
                Logs and errors: {BasicConfig.expid_aslog_dir(exp_id)}
                Attached to this message you will find the related _run.log files.
 
-               Platform affected: {str(platform.name)} using as host: {str(platform.host)}\n\n[WARN] Autosubmit encountered an issue with a remote platform.\n It will resume itself, whenever is possible\n If this issue persists, you can change the host IP or put multiple hosts in the platform.yml file' + '\n\n\n\n\nINFO: This message was auto generated by Autosubmit,
+               Platform affected: {platform.name} using as host: {platform.host}\n\n[WARN] Autosubmit encountered an issue with a remote platform.\n It will resume itself, whenever is possible\n If this issue persists, you can change the host IP or put multiple hosts in the platform.yml file' + '\n\n\n\n\nINFO: This message was auto generated by Autosubmit,
                 remember that you can disable these messages on Autosubmit config file.\n'''
diff --git a/test/unit/test_mail.py b/test/unit/test_mail.py
index a87ce3b3..b4ddc811 100644
--- a/test/unit/test_mail.py
+++ b/test/unit/test_mail.py
@@ -100,7 +100,8 @@ def test_notify_experiment_status(mail_notifier, mock_glob_side_effect, send_mai
                     else:
                         mail_notifier.notify_experiment_status(exp_id, mail_to, platform)
 
-                    mail_notifier._generate_message_experiment_status.assert_called_once_with(exp_id, platform)
+                    mail_notifier._generate_message_experiment_status.assert_called_once_with(  # type: ignore
+                        exp_id, platform)
                     mock_send_mail.assert_called_once_with(mail_notifier.config.MAIL_FROM, 'recipient@example.com', mock.ANY)
 
                     if mock_glob_side_effect is None: mock_attach_files.assert_called_once_with(mock.ANY, [file1, file2]) 
```

Refs:

- https://docs.python.org/3/library/typing.html
- https://peps.python.org/pep-0484/
- https://realpython.com/python-type-checking/

Reply:
Since `TYPE_CHECKING` is `False` at runtime (and testing), shouldn't I import it normally?